### PR TITLE
Address being able to friend yourself

### LINKED ISSUE #771: BUG: Players can friend themselves
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

Latest version

### Issue Description

Players should not be able to friend themselves but are currently able to by abusing the GM tag when adding a friend.

### Reproduction Steps

1. Open your friends list.
2. Click the +
3. Type [GM]yourname (with yourname being whatever your current player name is)
4. Notice that you get a friend request from yourself.

### Expected Behavior

Players should still get a message telling them they can't add themselves.

### Environment

Windows.

### PR DESCRIPTION
**Pull request by EmosewaMC**
Fix an issue where players could friend themselves.  Also stops yourself as appearing as a friend on your own friends list.

Tested adding myself as a friend using the [GM] workaround.  Was not able to.  Tested logging in and our and I did not show up as a friend on my own friend list.

Fixes #771

### CODE REVIEW

File: dChatServer/ChatPacketHandler.cpp [hunk 1]
```diff
@@ -113,6 +114,7 @@ void ChatPacketHandler::HandleFriendRequest(Packet* packet) {
 	inStream.Read(isBestFriendRequest);
 
 	auto requestor = playerContainer.GetPlayerData(requestorPlayerID);
+	if (requestor->playerName == playerName) return;
```
Comment (aronwk-aaron) [inline dChatServer/ChatPacketHandler.cpp:117]: Not blocking, but would it make sense to have this return a general error to the client, for the sake of it?

**Review by aronwk-aaron — APPROVED** [review]
just one comment/question, non-blocking

File: dChatServer/ChatPacketHandler.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dChatServer/ChatPacketHandler.cpp:117]: good idea.  Should implement that as well.

**Review by DarwinAnim8or — APPROVED** [review]
looks good

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**